Neos.Utility.Arrays/Tests/Unit/ArraysTest.php

```diff
@@ -108,6 +108,103 @@ public function getValueByPathReturnsNullIfThePathHasMoreSegmentsThanTheGivenArr
+    public function getAccessorByPathReturnsTheValueOfANestedArrayByFollowingTheGivenPath()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertSame('the value', Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 2])->string());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsTheValueOfANestedArrayByFollowingTheGivenPathIfPathIsString()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertSame('the value', Arrays::getAccessorByPath($array, 'Foo.Bar.Baz.2')->string());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathThrowsTypeErrorIfPathIsNoArrayOrString()
+    {
+        $this->expectException(\TypeError::class);
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        Arrays::getAccessorByPath($array, null);
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsNullIfTheSegementsOfThePathDontExist()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Bax', 2])->intOrNull());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsNullIfThePathHasMoreSegmentsThanTheGivenArray()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => 'the value']]];
+        self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 'Bux'])->intOrNull());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathTypeErrorContainsPathForNonMatchingTypes()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => 'the value']]];
+        try {
+            self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz'])->int());
+        } catch (\Error $error) {
+            self::assertInstanceOf(\TypeError::class, $error);
+            self::assertStringEndsWith('in path Foo.Bar.Baz', $error->getMessage());
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathTypeErrorContainsPathForNonMatchingTypesOnRoot()
+    {
+        $array = ['Foo' => 'string'];
+        try {
+            self::assertNull(Arrays::getAccessorByPath($array, ['Foo'])->int());
+        } catch (\Error $error) {
+            self::assertInstanceOf(\TypeError::class, $error);
+            self::assertStringEndsWith('in path Foo', $error->getMessage());
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathTypeErrorContainsPathForNonExistingPathed()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => 'the value']]];
+        try {
+            self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 'Bux'])->intOrNull());
+        } catch (\Error $error) {
+            self::assertInstanceOf(\TypeError::class, $error);
+            self::assertStringEndsWith('in path Foo.Bar.Baz.Bux', $error->getMessage());
+        }
```

Comment:
No exception is thrown in this case (another reason not to use `try/catch` in those tests)
```suggestion
        self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 'Bux'])->intOrNull());
```

Reply:
adjusted as requested